src/modelinstance.cpp

```diff
@@ -632,6 +632,14 @@ void ModelInstance::unloadModel(bool isPermanent) {
     }
 }
 
+const bool ModelInstance::checkIfShapeValuesNegative(const tensorflow::TensorProto& requestInput) {
```

Comment:
Why using bool if you can return ret code from here with details provided & log the message without bloating validate method?

Then just 
status = checkIfShapeValuesNegative(requiestInput);
if (!status.ok() ) {
    return status;